### PR TITLE
Add PlacementConfiguration to kubeMacPool

### PR DESCRIPTION
**Pull request by rhrazdil**
Add Infra Placement configuration to kubeMacPool taken from NetworkAddonsConfig CR.

Adding Helm chartutil dependency which provides toYaml method for
expanding affinity/tolerations/nodeSelector structures in the
go templates.

Depends on https://github.com/kubevirt/cluster-network-addons-operator/pull/520

Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RamLavi** [comment]
@rhrazdil, please split vendor changes to a different commit, to make reviewing easier

File: hack/components/bump-kubemacpool.sh [hunk 1]
```diff
@@ -83,7 +86,10 @@ rm -rf data/kubemacpool/*
     ./build/_output/bin/go/bin/kustomize build config/cnao | \
         sed 's/kubemacpool-system/{{ .Namespace }}/' | \
         sed 's/RANGE_START: .*/RANGE_START: {{ .RangeStart }}/' | \
-        sed 's/RANGE_END: .*/RANGE_END: {{ .RangeEnd }}/'
+        sed 's/RANGE_END: .*/RANGE_END: {{ .RangeEnd }}/' | \
+        sed 's/AFFINITY/{{ toYaml .Placement.Affinity | nindent 8 }}/' | \
```
Comment (RamLavi) [inline hack/components/bump-kubemacpool.sh:90]: IIUC you're only seding this in order to remove the `''` right?

Comment (rhrazdil) [reply hack/components/bump-kubemacpool.sh:90]: Yes, expanding the structure with surrounding quotes leads to invalid yaml

**Comment by phoracek** [comment]
Travis fail:

```
Fail] Testing internal git getUpdatedReleaseInfo function [It] Update-policy tagged: should return latest tag in release branch. current: v1.0.0 
```

**Comment by phoracek** [comment]
It is unrelated. I think we can blame @RamLavi

File: hack/components/bump-kubemacpool.sh [hunk 2]
```diff
@@ -45,6 +45,9 @@ metadata:
 spec:
   template:
     spec:
+      affinity: AFFINITY
```
Comment (phoracek) [inline hack/components/bump-kubemacpool.sh:48]: This patch is called "cnao_image_patch". Would you please introduce a new patch for placement?

**Review by phoracek — CHANGES_REQUESTED** [review]
Just a nit. Other than that, I tried the PR locally and its smooth! Thanks

File: hack/components/bump-kubemacpool.sh (hunk 2 above)
Comment (rhrazdil) [reply hack/components/bump-kubemacpool.sh:48]: Ah, I see! Fixed

Comment (rhrazdil) [reply hack/components/bump-kubemacpool.sh]: ops, nope. mistake was made

Comment (rhrazdil) [reply hack/components/bump-kubemacpool.sh]: now it's fixed

**Review by phoracek — APPROVED** [review]
Cool, thanks a lot!

/lgtm
/approve

**Pull request merged**